tools/integration-tests/tester/tests/testutil.go

```diff
@@ -231,6 +233,59 @@ func SendFaucetRequest(t *testing.T, node *framework.Node, addr ledgerstate.Addr
 	return resp.ID, sent
 }
 
+// region CreateTransaction from outputs //////////////////////////////
+
+/**
+CreateTransactionFromOutputs takes the given utxos inputs and create a transaction tnat spread the total input balance
+across the targetAddresses. In order to correctly sign we have akeyPair map that maps a given address to its public key.
+Access and Consensus Mana is pledged to the node we specify.
+*/
+func CreateTransactionFromOutputs(t *testing.T, manaPledgeID identity.ID, targetAddresses []*ledgerstate.Address, keyPairs map[string]*ed25519.KeyPair, utxos ...ledgerstate.Output) *ledgerstate.Transaction {
```

Comment:
Is it really necessary to pass the map of addresses to keypair here? Why cannot this function itself get the keypair? It would hide some complexity from the calling functions.

Reply:
how would it do it?  I create the keypairs together with the addresses. After the addresses are created I am not sure how I can get the corresponding keypairs. Unless I save the index for each address and poass it down but it is also ugly to me.

In most tests we use the wallet, but in this test I wanted to have a finer grain control on the outputs than the wallet allows. It was mainly because initially I was tryiong to do even fancier stuff that didn't work out for the test. 
Maybe think of the map and outputs as an `OutputWallet`?

I can also create an OutputWallet struct that contains all the outputs and their keys (combining 2 variables here).
Not sure if this is what you have in mind so I'll be waiting for your answer.

Reply:
If I understand correctly, this function is meant to distribute the balances of the passed UTXOs onto target addresses. But why do you need a address->keypair map to create the new transactions?
In the caller function you actually know which peer controls which address when you create the outputs in the first place. Can this function receive that peer instead so you avoid to pass keypairs all the way through?

Reply:
It is not this simple unfortunately.

This is because in order to derive the correct key from an address I need it's index.
So I will need to pass down the indexes of each address. That will also be ugly.
I think it is better to pass down the map. Maybe give it another name?
I can also do that `OutputWallet` idea that will at least force to have a key associated with the output.